### PR TITLE
247 appd use of interop

### PR DESCRIPTION
**Pull request by kriswest**
resolves https://github.com/finos/FDC3/issues/247

Adds a set of fields to AppD application records to describe an app's use of FDC3 and enable search for apps that 'interoperate' with a selected app.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -30,6 +30,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 * Added `localizedVersions` field to AppD application records to support localized versions of descriptive fields in the app records and alternative launch details for localized versions of the applications themselves. ([#670](https://github.com/finos/FDC3/pull/670))
 * Added `type` and `details` elements to AppD application records to support vendor-agnostic launch details for both web and native apps ([#671](https://github.com/finos/FDC3/pull/671))
 * Added `categories` field and recommended categories list to AppD application records to enable category based browsing of AppDs ([#673](https://github.com/finos/FDC3/pull/673))
+* Added `interopUse` field to AppD application records to describe an app's use of FDC3 and enable search for apps that 'interoperate' with a selected app ([#697](https://github.com/finos/FDC3/pull/697))
```
Comment (kriswest) [inline CHANGELOG.md:33]:
```suggestion
* Added `raiseIntent`, `raiseIntentForContext`, appChannels`, and `userChannels` fields to AppD application records to describe an app's use of FDC3 and enable search for apps that 'interoperate' with a selected app ([#697](https://github.com/finos/FDC3/pull/697))
```

**Comment by kriswest** [comment]
After a couple of false starts, this now has a preview at: https://deploy-preview-731--lambent-kulfi-cf51a7.netlify.app/schemas/next/app-directory#tag/Application/paths/~1v2~1apps~1{appId}/get

image

I'll add an example that uses it before release (not FDC3Workbench as its use is dynamic making it a poor example for these fields). 

@nkolba do you prefer this to the previous version (with an `interopUse` field that wrapped these fields). I think it better now that they are siblings of `intents`.

**Review by greyseer256 — APPROVED** [review]
This all looks fine, just not sure about the capitalization of 'application'

File: src/app-directory/specification/appd.yaml [hunk 2]
```diff
@@ -784,6 +795,81 @@ components:
       additionalProperties:
         x-additionalPropertiesName: Language tag
         $ref: '#/components/schemas/BaseApplication' # due to a bug in redoc this may display as a recursive definition, it is not. It will render correctly in swagger and other OpenAPI parsers.
+    RaiseIntent:
+      type: object
+      description: >-
+        Optional metadata that describes which Intents are raised by the application and context types they are
```
[the remark this answers is not in the document]
Comment (kriswest) [reply src/app-directory/specification/appd.yaml]: Fixed. Lowercase in content, uppercase in Application Directory

**Comment by kriswest** [comment]
Latest preview with examples: https://deploy-preview-731--lambent-kulfi-cf51a7.netlify.app/schemas/next/app-directory#tag/Application/

@nkolba

**Review by mattjamieson — APPROVED** [review]

**Comment by kriswest** [comment]
I've had a request to rename `interopMeta` to `interop`, which I'm fine with and actually prefer (as the whole record is metadata meta seems a little redundant). I've checked in with @nkolba on this (who suggested the current name) who is also fine with it (shorter names are good). 

The original intention was to reinforce the non-binding aspects of this metadata (although the intent listening metadata is binding, the rest is not directly used by the desktop agent). 

Hence, if there are no objections, I'll update the PR with this small change.

**Comment by kriswest** [comment]
Approved via email vote:
- https://groups.google.com/a/finos.org/g/fdc3/c/gKQ-Jdnqju0/m/b65z8t3WAQAJ

Minor change to the name of the element to be applied (`interopMeta` -> `interop`).

**Pull request merged**